### PR TITLE
feature/327

### LINKED ISSUE #327: New Airport: KABQ
Albuquerque International Sunport (IATA: ABQ, ICAO: KABQ, FAA LID: ABQ) is a public airport 3 miles (5 km) southeast of downtown Albuquerque, in Bernalillo County, New Mexico, United States. It is the largest commercial airport in the state, handling 4,745,256 passengers in 2015. The airport serves Albuquerque and Santa Fe.

### PR DESCRIPTION
**Pull request by Fechulo**
Resolves #327.

Add KABQ.

### CODE REVIEW

File: assets/airlines/ash.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "icao": "ash",
```
Comment (indianbhaji) [inline assets/airlines/ash.json]: 4 space indenting

File: assets/airports/kabq.json [hunk 2]
```diff
@@ -0,0 +1,1986 @@
+{
+  "radio": {
```
Comment (indianbhaji) [inline assets/airports/kabq.json]: 4 space indenting

File: assets/airports/kabq.json [hunk 3]
```diff
@@ -0,0 +1,1986 @@
...
+    "YUGLU": ["N34.94116718226414", "W106.51361846469737"],
+    "YYLEE": ["N34.95837118211725", "W106.21692069811836"],
+    "ZATOX": ["N34.83574265046455", "W104.61931698566244"],
+    "ZUN":   ["N34.96576096364239", "W109.15452575246168"]
+  },
+  "runways":[
+    {
+      "name": ["3", "21"],
+      "end": [["N35d01m20.09", "W106d37m50.17", "5305ft"], ["N35d02m30.27", "W106d36m25.46", "5316ft"]],
+      "delay": [8, 5],
```
Comment (indianbhaji) [inline assets/airports/kabq.json]: `delay` not in use I believe?

**Comment by Fechulo** [comment]
Ready for review 👍

File: assets/airports/kabq.json [hunk 4]
```diff
@@ -0,0 +1,2105 @@
...
+            "exitPoints": {
+                "ONM": ["DUMMM", "ONM"],
+                "SJN": ["WUTUP", "LAMSE", "SJN"],
+                "CME": ["DADDY", "LARLE", "TACOZ", "CME"],
+                "TXO": ["FATTT", "MOLVE", "TXO"]
+            },
+            "draw": [
+                ["DSPOT", "JETOK"],
+                ["ONM*"],
+                ["SJN*"],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: It isn't really able to draw a line with only one point, so these should be like:
```
"draw": [
    ["DSPOT", "JETOK"],
    ["JETOK", "DUMMM", "ONM*"],
    ["JETOK", "WUTUP", "LAMSE", "SJN*"],
    ["JETOK", "DADDY", "LARLE", "TACOZ", "CME*"],
    ["JETOK", "FATTT", "MOLVE", "TXO*"]
]
```

File: assets/airports/kabq.json [hunk 5]
```diff
@@ -0,0 +1,2105 @@
...
+                ["DLSOL", "YUGLU", "CIANE", "MNZNO", "PEPTE", "WUTUP"],
+                ["MNZNO", "HRDRK", "DUMMM"],
+                ["MNZNO", "DADDY"],
+                ["MNZNO", "FATTT"],
+                ["MOLVE*"],
+                ["LAMSE*"],
+                ["ONM*"],
+                ["SJN*"],
+                ["CME*"],
+                ["TXO*"]
```
Comment (erikquinn) [inline assets/airports/kabq.json]: need multiple points to draw blue lines.

File: assets/airports/kabq.json [hunk 6]
```diff
@@ -0,0 +1,2105 @@
...
+            "body": [["RDRNR", "A100+"], ["SPEDY", "A120+"]],
+            "exitPoints": {
+                "GUP": ["BABYY", "BUSLE", "GUP"],
+                "ZUN": ["PNYON", "ZUN"]
+            },
+            "draw": [
+                ["TYILR", "RDRNR", "SPEDY", "BABYY"],
+                ["SPEDY", "PNYON"],
+                ["GUP*"],
+                ["ZUN*"]
```
Comment (erikquinn) [inline assets/airports/kabq.json]: can't do singles etc (more examples throughout)

File: assets/airports/kabq.json [hunk 7]
```diff
@@ -0,0 +1,2105 @@
...
+            "icao": "LZZRD3",
+            "name": "Lizzard Three",
+            "entryPoints": {
+                "CME": ["CME", "LOKKE", ["JETOS", "A260+|A300-|S280"], "BOBYY", "HAKOR"]
+            },
+            "body": [["LZZRD", "A140+|A180-|S250"], ["MKYON", "A130+|A150-"]],
+            "rwy": {
+                "8": [["HAPEE", "A110+|A130-"], ["COMRO", "A100|S210"]],
+                "3": [["HAPEE", "A110+|A130-"], ["COMRO", "A100|S210"]],
+                "21": ["SOPPA", ["PILLA", "A100|S210"]],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Please reorder runway `3` and `8` entries

File: assets/airports/airportLoadList.js [hunk 8]
```diff
@@ -150,6 +150,12 @@ window.AIRPORT_LOAD_LIST = (function() { // eslint-disable-line wrap-iife
             wip: true
         },
         {
+            icao: 'kabq',
+            level: 'easy',
+            name: 'Albuquerque International Sunport',
+            wip: true
```
Comment (erikquinn) [inline assets/airports/airportLoadList.js]: `false`

File: assets/airports/kabq.json [hunk 9]
```diff
@@ -0,0 +1,2105 @@
...
+        {
+            "name": ["17", "35"],
+            "end": [
+                ["N35d03m27.83", "W106d36m39.6", "5312ft"],
+                ["N35d01m49.12", "W106d36m46.68", "5314ft"]
+            ],
+            "ils": [false, false]
+        }
+    ],
+    "sids": {
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Please add `LARGO2` and `ABQ2` SIDs.

File: assets/airports/kabq.json (hunk 4 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: The problem is these lines are already being drawn in other procedures. The single fix lines, are just drawing the name of the procedure at the exitPoints, and it works fine:
capture
The line from DUMMM to ONM is only being drawn once, but the all the names of all the different procedures that end at ONM are there.

Comment (erikquinn) [inline assets/airports/kabq.json]: Ahhhhh I'm with ya. Doesn't seem to break anything, which is interesting. I can roll with it I suppose. :+1:

File: assets/airports/kabq.json (hunk 9 above)
Comment (erikquinn) [inline assets/airports/kabq.json]: done

File: assets/airports/kabq.json [hunk 10]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "FYSTA3": {
+            "icao": "FYSTA3",
+            "name": "Fiesta Three",
+            "rwy": {
+                "21": [["^_RWY0301.27DME", "A62+"]],
+                "26": [["^_RWY0803.22DME", "A75+"]]
+            },
+            "body": [["BLUUN", "A110+|S250-"], ["BURQI", "A120+|S250-"], ["FYSTA", "A140+"]],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: no speed restriction at `BLUUN`

File: assets/airports/kabq.json [hunk 11]
```diff
@@ -0,0 +1,2170 @@
...
+                ["JEMEZ"]
+            ]
+        },
+        "JETOK3": {
+            "icao": "JETOK3",
+            "name": "Jeetock Three",
+            "rwy": {
+                "3": []
+            },
+            "body": [["^_RWY2101.50DME", "A64+|S210-"], ["DSPOT", "A90-|S210-"], ["JETOK", "A100+"]],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: 210 speed restriction should not be on the `_RWY2102.50DME` fix.
`DSPOT` should be `A90+|S210-`

File: assets/airports/kabq.json [hunk 12]
```diff
@@ -0,0 +1,2170 @@
...
+            "exitPoints": {
+                "ONM": ["DUMMM", "ONM"],
+                "SJN": ["WUTUP", "LAMSE", "SJN"],
+                "CME": ["DADDY", "LARLE", "TACOZ", "CME"],
+                "TXO": ["FATTT", "MOLVE", "TXO"]
```
Comment (erikquinn) [inline assets/airports/kabq.json]: sort alphabetically

File: assets/airports/kabq.json [hunk 13]
```diff
@@ -0,0 +1,2170 @@
...
+                ["TXO*"]
+            ]
+        },
+        "LARGO2": {
+            "icao": "LARGO2",
+            "name": "Largo Two",
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Isn't this also valid for 12-30?

File: assets/airports/kabq.json [hunk 14]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "MNZNO3": {
+            "icao": "MNZNO3",
+            "name": "Manzano Three",
+            "rwy": {
+                "3": [["^_RWY2101DME", "A60+|S220-"]],
+                "8": []
+            },
+            "body": ["DLSOL", ["YUGLU", "A90-"], ["CIANE", "S250-"], ["MNZNO", "A115+|S250-"]],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: no speed restriction at `CIANE`

File: assets/airports/kabq.json [hunk 15]
```diff
@@ -0,0 +1,2170 @@
...
+            "exitPoints": {
+                "ONM": [["HRDRK", "A160+"], "DUMMM", "ONM"],
+                "SJN": [["PEPTE", "A150+"], ["WUTUP", "A160+"], "LAMSE", "SJN"],
+                "LAMSE": [["PEPTE", "A150+"], ["WUTUP", "A160+"], "LAMSE"],
+                "CME": ["DADDY", "LARLE", "TACOZ", "CME"],
+                "MOLVE": ["FATTT", "MOLVE"],
+                "TXO": ["FATTT", "MOLVE", "TXO"]
```
Comment (erikquinn) [inline assets/airports/kabq.json]: sort alphabetically

File: assets/airports/kabq.json [hunk 16]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "CURLY3": {
+            "icao": "CURLY3",
+            "name": "Curly Three",
+            "entryPoints": {
+                "GUP": ["GUP"],
+                "TANER": ["TANER"]
+            },
+            "body": [["CURLY", "S300-"], "ABQ"],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: no speed restriction at `CURLY`

File: assets/airports/kabq.json [hunk 17]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "FRIHO4": {
+            "icao": "FRIHO4",
+            "name": "Freeoh Four",
+            "entryPoints": {
+                "ESPAN": ["ESPAN", "FLYBY"],
+                "FTI": ["FTI", "FLYBY"]
+            },
+            "body": [["FRIHO", "S300-"], "ABQ"],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: We can move `FLYBY` to the body
No speed restriction at `FRIHO`

File: assets/airports/kabq.json [hunk 18]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "LAVAN3": {
+            "icao": "LAVAN3",
+            "name": "Lahvan Three",
+            "entryPoints": {
+                "SJN": ["SJN", "IRULY"],
+                "ZUN": ["ZUN", "IRULY"]
+            },
+            "body": [["LAVAN", "S300-"], "ABQ"],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: `IRULY` can be moved to the body here
`LAVAN` has no restrictions

File: assets/airports/kabq.json [hunk 19]
```diff
@@ -0,0 +1,2170 @@
...
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Isn't this also valid for 12-30?

File: assets/airports/kabq.json [hunk 20]
```diff
@@ -0,0 +1,2170 @@
...
+            "name": "Curly Three",
+            "entryPoints": {
+                "GUP": ["GUP"],
+                "TANER": ["TANER"]
+            },
+            "body": [["CURLY", "S300-"], "ABQ"],
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Isn't this also valid for 12-30?

File: assets/airports/kabq.json [hunk 21]
```diff
@@ -0,0 +1,2170 @@
...
+            "name": "Freeoh Four",
+            "entryPoints": {
+                "ESPAN": ["ESPAN", "FLYBY"],
+                "FTI": ["FTI", "FLYBY"]
+            },
+            "body": [["FRIHO", "S300-"], "ABQ"],
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Isn't this also valid for 12-30?

File: assets/airports/kabq.json [hunk 22]
```diff
@@ -0,0 +1,2170 @@
...
+            "entryPoints": {
+                "CME": ["CME", "CNX"],
+                "TXO": ["TXO"],
+                "TCC": ["TCC", "ACH"]
+            },
+            "body": [["MIERA", "S300-"], "ABQ"],
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: Isn't this also valid for 12-30?

File: assets/airports/kabq.json [hunk 23]
```diff
@@ -0,0 +1,2170 @@
...
+        },
+        "MIERA2": {
+            "icao": "MIERA2",
+            "name": "Meera Two",
+            "entryPoints": {
+                "CME": ["CME", "CNX"],
+                "TXO": ["TXO"],
+                "TCC": ["TCC", "ACH"]
+            },
+            "body": [["MIERA", "S300-"], "ABQ"],
```
Comment (erikquinn) [inline assets/airports/kabq.json]: no speed restriction at `MIERA`

File: assets/airports/kabq.json [hunk 24]
```diff
@@ -0,0 +1,2170 @@
...
+            ]
+        },
+        "ATOMK3": {
+            "icao": "ATOMK3",
+            "name": "Atomkay Three",
+            "rwy": {
+                "21": [],
+                "26": []
+            },
+            "body": [["FOXRR", "A80+"], ["TYWEB", "A100+|S250-"], ["TKILA", "A115+|S250-"], "FIREY"],
```
Comment (Fechulo) [inline assets/airports/kabq.json]: Regarding this restriction and the other ones in the SID's: They're there to make sure that aircraft don't speed up above 250 at TYWEB and then slow down for the actual restriction at TKILA. The restriction at TKILA also applies to all the fixes before it but last time I checked aircraft ignored that (Maybe it works now with all the changes made to VNAV?). Hence the restriction at TYWEB

File: assets/airports/kabq.json [hunk 25]
```diff
@@ -0,0 +1,2170 @@
...
+        "twr": "Albuquerque Tower",
+        "app": "Albuquerque Approach",
+        "dep": "Albuquerque Departure"
+    },
+    "icao": "KABQ",
+    "iata": "ABQ",
+    "magnetic_north": 8.8,
+    "ctr_radius": 80,
+    "ctr_ceiling": 20000,
+    "initial_alt": 10000,
```
Comment (Fechulo) [inline assets/airports/kabq.json]: Done.

File: assets/airports/kabq.json [hunk 26]
```diff
@@ -0,0 +1,2170 @@
...
+                ["ABQ", "ADYOS", "PNYON", "ZUN*"]
+            ]
+        },
+        "ABQ2": {
+            "icao": "ABQ2",
+            "name": "Albuquerque Two",
+            "rwy": {
+                "3": [],
+                "8": [],
+                "21": [],
```
Comment (Fechulo) [inline assets/airports/kabq.json]: Done.

File: assets/airports/kabq.json [hunk 27]
```diff
@@ -0,0 +1,2170 @@
...
+                "26": []
+            },
+            "body": [],
+            "exitPoints": {
+                "ALS": ["ALS"],
```
Comment (Fechulo) [inline assets/airports/kabq.json]: Done.

File: assets/airports/kabq.json (hunk 12 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: Done.

File: assets/airports/kabq.json (hunk 13 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: Done.

File: assets/airports/kabq.json (hunk 15 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: All Fixed.

File: assets/airports/kabq.json (hunk 16 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: All Fixed.

File: assets/airports/kabq.json (hunk 20 above)
Comment (Fechulo) [inline assets/airports/kabq.json]: All Fixed.

File: assets/airports/kabq.json (hunk 24 above)
Comment (erikquinn) [inline assets/airports/kabq.json]: I figured that was the reason, but regardless it should be implemented as published. If there's an issue with the sim I'd rather face and fix the problem than hide it by fudging an airport slightly. And yes, I believe it should now be working correctly anyway.

Comment (Fechulo) [inline assets/airports/kabq.json]: Makes sense 👍

File: assets/airports/kabq.json (hunk 10 above)
Comment (erikquinn) [inline assets/airports/kabq.json]: done

File: assets/airports/kabq.json (hunk 11 above)
Comment (erikquinn) [inline assets/airports/kabq.json]: done

**Comment by Fechulo** [comment]
Sounds good! 😄

**Pull request merged**